kore/src/Kore/Step/Step.hs

```diff
@@ -296,6 +295,8 @@ applyRemainder initial remainder = do
             remainder
     -- Add the simplified remainder to the initial conditions. We must preserve
     -- the initial conditions here!
-    Simplifier.simplifyCondition
-        SideCondition.topTODO
-        (Pattern.andCondition initial partial)
+    r <-
+        Simplifier.simplifyCondition
+            SideCondition.topTODO
+            (Pattern.andCondition initial partial)
+    return (r & Pattern.mapVariables resetConfigVariable)
```

Comment:
Could this be cleaned up a bit? I hope this compiles, if not, maybe we should at least give a more descriptive name to `r`.

```suggestion
    Simplifier.simplifyCondition
        SideCondition.topTODO
        (Pattern.andCondition initial partial)
        <&> Pattern.mapVariables resetConfigVariable
```